client.go

```diff
@@ -163,6 +182,13 @@ func NewClient(opts ...ClientOption) (*Client, error) {
 		client.imageFactory = image.NewFactory(client.docker, authn.DefaultKeychain)
 	}
 
+	if client.assetFetcher == nil {
+		localAssetFetcher := asset.NewPackageFileFetcher()
+		imageAssetFetcher := asset.NewPackageImageFetcher(client.imageFetcher)
```

Comment:
I find this line pretty odd, given that `client.assetFetcher` is `nil` in this block.

Reply:
Could you elaborate? I'm having trouble understanding whats off here.

Reply:
Ah never mind, I thought you were passing `client.assetFetcher` into the function, not `client.imageFetcher`